src/bin/lpython.cpp

```diff
@@ -449,6 +449,151 @@ int get_errors (const std::string &infile,
+        uint32_t end_character = loc.last_column;
+        uint32_t end_line = loc.last_line;
+        std::string name = loc.symbol_name;
+
+        range_object.SetObject();
+        rapidjson::Document::AllocatorType &allocator =
+            range_object.GetAllocator();
+
+        start_detail.SetObject();
+        start_detail.AddMember(
+            "character", rapidjson::Value().SetInt(start_character), allocator);
+        start_detail.AddMember("line", rapidjson::Value().SetInt(start_line),
+                               allocator);
+        range_object.AddMember("start", start_detail, allocator);
+
+        end_detail.SetObject();
+        end_detail.AddMember("character",
+                             rapidjson::Value().SetInt(end_character), allocator);
+        end_detail.AddMember("line", rapidjson::Value().SetInt(end_line),
+                             allocator);
+        range_object.AddMember("end", end_detail, allocator);
+
+        location_object.SetObject();
+        location_object.AddMember("range", range_object, allocator);
+        location_object.AddMember(
+            "uri", rapidjson::Value().SetString("uri", allocator), allocator);
+
+        test_capture.SetObject();
+        test_capture.AddMember("kind", rapidjson::Value().SetInt(1), allocator);
+        test_capture.AddMember("location", location_object, allocator);
+        test_capture.AddMember(
+            "name", rapidjson::Value().SetString(name.c_str(), allocator),
+            allocator);
+        test_output.PushBack(test_capture, test_output.GetAllocator());
+        rapidjson::StringBuffer buffer;
+        buffer.Clear();
+        rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+        test_output.Accept(writer);
+        std::string resp_str(buffer.GetString());
+
+        std::cout << resp_str;
+        return true;
+      }
+    }
+    std::cout << "{}\n";
+    return false;
+  }
+  std::cout << "{}\n";
+  return false;
+}
+
+std::pair<std::pair<int, int>, std::pair<int, int>>
+return_goto_def(const std::string &infile,
+                const std::string &runtime_library_dir,
+                CompilerOptions compiler_options, int index) {
+  std::string input = LFortran::read_file(infile);
+  // Src -> Tokens
+  Allocator al(64 * 1024 * 1024);
+  std::vector<int> toks;
+  std::vector<LFortran::YYSTYPE> stypes;
+  std::vector<LFortran::Location> locations;
+  LFortran::diag::Diagnostics diagnostics;
+  auto res = LFortran::tokens(al, input, diagnostics, &stypes, &locations);
+  LFortran::LocationManager lm;
+  lm.in_filename = infile;
+  lm.init_simple(input);
+  // std::cerr << diagnostics.render(input, lm, compiler_options);
+  if (res.ok) {
+    toks = res.result;
+  } else {
+    std::cout << "{}\n";
+    return {};
+  }
+
+  for (size_t i = 0; i < toks.size(); ++i) {
+    if (input_in_scope_identifier(
+            index,
+            std::make_pair<int, int>(locations[i].first, locations[i].last))) {
```

Comment:
I would just do:
```suggestion
    if (locations[i].first <= index && index <= locations[i].last) {
}
```
And remove `input_in_scope_identifier`.